src/gleam/uri.gleam

```diff
@@ -119,6 +119,36 @@ pub fn query_to_string(query: List(tuple(String, String))) -> String {
   |> result.unwrap("")
 }
 
+/// Encode a string into a percent encoded representation.
+/// Note that this encodes space as +.
+///
+/// ## Example
+///
+/// percent_encode("100% great")
+/// > "100%25+great"
+///
+pub fn percent_encode(value: String) -> String {
+  query_to_string([tuple("k", value)])
+  |> string.replace(each: "k=", with: "")
+}
+
+/// Decode a percent encoded string.
+///
+/// ## Example
+///
+/// percent_decode("100%25+great")
+/// > Ok("100% great")
+///
+pub fn percent_decode(value: String) -> Result(String, Nil) {
+  parse_query(string.concat(["k=", value]))
+  |> result.then(fn(key_values) {
+    case list.head(key_values) {
+      Ok(tuple(_, v)) -> Ok(v)
+      _ -> Error(Nil)
+    }
+  })
+}
```

Comment:
Nitpick: Perhaps we could write this like this? It's the same, but maybe a little neater.
```suggestion
  string.concat(["k=", value])
  |> parse_query
  |> result.then(list.head)
  |> result.map(pair.second)
```

Reply:
Done, much nicer thanks